### PR TITLE
[Merged by Bors] - feat(Algebra): add `IsLocalizedModule.liftOfLE`

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/LocalizedModule.lean [hunk 1]
```diff
@@ -1037,6 +1037,52 @@ theorem mk'_surjective : Function.Surjective (Function.uncurry <| mk' f : M × S
...
+section liftOfLE
+
+variable {M₁ M₂} [AddCommMonoid M₁] [AddCommMonoid M₂] [Module R M₁] [Module R M₂]
+variable (S₁ S₂ : Submonoid R) (h : S₁ ≤ S₂) (f₁ : M →ₗ[R] M₁) (f₂ : M →ₗ[R] M₂)
+variable [IsLocalizedModule S₁ f₁] [IsLocalizedModule S₂ f₂]
+
+/-- The natural map `Mₛ →ₗ[R] Mₜ` if `s ≤ t` (in `Submonoid R`). -/
+noncomputable
+def liftOfLE : M₁ →ₗ[R] M₂ :=
+  lift S₁ f₁ f₂ fun x ↦ map_units f₂ ⟨x.1, h x.2⟩
```
Comment (chrisflav) [inline Mathlib/Algebra/Module/LocalizedModule.lean:1049]: Can the `(S₁ S₂ : Submonoid R)` be implicit here?

File: Mathlib/Algebra/Module/LocalizedModule.lean [hunk 2]
```diff
@@ -1037,6 +1037,52 @@ theorem mk'_surjective : Function.Surjective (Function.uncurry <| mk' f : M × S
...
+
+/-- The natural map `Mₛ →ₗ[R] Mₜ` if `s ≤ t` (in `Submonoid R`). -/
+noncomputable
+abbrev _root_.LocalizedModule.liftOfLE : LocalizedModule S₁ M →ₗ[R] LocalizedModule S₂ M :=
+  IsLocalizedModule.liftOfLE S₁ S₂ h
+    (LocalizedModule.mkLinearMap S₁ M) (LocalizedModule.mkLinearMap S₂ M)
+
+lemma liftOfLE_comp : (liftOfLE S₁ S₂ h f₁ f₂).comp f₁ = f₂ := lift_comp ..
```
Comment (chrisflav) [inline Mathlib/Algebra/Module/LocalizedModule.lean:1057]: Wow, what do the `..` dots do? I have never seen that before.

**Review by chrisflav — APPROVED** [review]
Thanks!

File: Mathlib/Algebra/Module/LocalizedModule.lean (hunk 1 above)
Comment (erdOne) [reply Mathlib/Algebra/Module/LocalizedModule.lean:1049]: I don't think we usually have the inequality floating around so I think we should keep S explicit here.

File: Mathlib/Algebra/Module/LocalizedModule.lean (hunk 2 above)
Comment (erdOne) [reply Mathlib/Algebra/Module/LocalizedModule.lean:1057]: It is essentially "add `_` until you can't"

File: Mathlib/Algebra/Module/LocalizedModule.lean (hunk 1 above)
Comment (chrisflav) [reply Mathlib/Algebra/Module/LocalizedModule.lean:1049]: Also fine.

**Comment by chrisflav** [comment]
Thanks!

maintainer merge

File: Mathlib/Algebra/Module/LocalizedModule.lean (hunk 2 above)
Comment (chrisflav) [reply Mathlib/Algebra/Module/LocalizedModule.lean:1057]: Aha, thanks

File: Mathlib/Algebra/Module/LocalizedModule.lean [hunk 3]
```diff
@@ -1037,6 +1037,52 @@ theorem mk'_surjective : Function.Surjective (Function.uncurry <| mk' f : M × S
...
+
+@[simp] lemma liftOfLE_apply (x) : liftOfLE S₁ S₂ h f₁ f₂ (f₁ x) = f₂ x := lift_apply ..
+
+set_option linter.docPrime false in
```
Comment (mattrobball) [inline Mathlib/Algebra/Module/LocalizedModule.lean:1061]: Why not add a short doc string explaining this is compatibility with `mk'`?

Comment (erdOne) [reply Mathlib/Algebra/Module/LocalizedModule.lean]: I feel like the name and the type signature are clear enough and the extra docstring is just distraction when scanning over this file looking for actually interesting results, but I added a short one anyways.

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**